### PR TITLE
Attestation Request/response spec (ch5490)

### PR DESCRIPTION
**Pull request by christianlundkvist**
Includes the attestation request/response spec, including plantuml diagrams.

### CODE REVIEW

**Comment by christianlundkvist** [comment]
@oed Personally I would prefer doing a separate spec for ERC712, it seems hard to pack both of them into the same spec.

**Comment by oed** [comment]
@christianlundkvist ok, that sounds good. Can we make that a priority?

File: messages/attreq.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+#### Attributes
+
+The JWT shares these attributes with the [Share Request](sharereq.md): `iss`, `iat`, `exp`.
+
+The following additional attributes of the JWT are supported:
+
+Name | Description | Required
+---- | ----------- | --------
+`type` | MUST have the value `attReq` | yes
+`unsignedClaims` | A list of unsigned claims that the user is requested to sign, in the form of JSON objects that form part of the payload of the attestation. A mandatory field is the `claim` itself, and a common field would be `sub` for the subject. | yes
```
Comment (oed) [inline messages/attreq.md]: I think we need to have an example of one of these!

File: flows/attestation.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+The client app MUST include a URL where the response is returned from the user. This can be a HTTPS URL or a custom app URL which receives the response.
+
+Responses are param appended to a URL fragment. If the callback requires the response as a HTTP POST, it is sent as a JSON POST request to the callback URL instead.
+
+### Successful Response
+
+param          | Description
+-------------- | -----------
+`access_token` | [Attestation Response](../messages/attresp.md)
```
Comment (simonas-notcat) [inline flows/attestation.md]: Does it make sense to call it `access_token` if it has nothing to do with _access_ ?

File: flows/attestation.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+The following shows the basic flow:
+
+![Attestation Flow](attestation.png)
+
+## Endpoint
+
+The request should be formatted using one of the following URLs:
+
+- `me.uport:me`
+- `https://id.uport.me/me`
```
[the remark this answers is not in the document]
Comment (ajunge) [reply flows/attestation.md]: m for message
* `me.uport:m`
* `https://id.uport.me/m?=....`

or  t for token?
* `me.uport:t`
* `https://id.uport.me/t?=....`

File: flows/attestation.md (hunk 2 above)
Comment (ajunge) [reply flows/attestation.md]: Shouldn't be called `r` (for `response token`)?

File: flows/verificiationreq.md [hunk 4]
```diff
@@ -10,8 +10,8 @@ The following shows the basic flow:
 
 The request should be formatted using one of the following URLs:
 
-- `me.uport:me`
-- `https://id.uport.me/me`
+- `me.uport:req`
```
Comment (christianlundkvist) [inline flows/verificiationreq.md]: We’re using `req` for all requests and distinguishing them by the `type` field in the JWT? Sounds good to me.

**Comment by christianlundkvist** [comment]
@pelle @oed @simonas-notcat Some great changes here! It’s more clear and well-defined now! A couple things I’d like to fix:

1. `sub` should be `rsub` to match the meaning of `rexp`
2. The description of this `rsub` needs to be updated.

I also like the way we are now asking the user to sign just a single claim rather than a list of claims (which we had before)

**Comment by pelle** [comment]
@christianlundkvist I think `sub` is still the correct one based on the JWT spec. It's the subject of the request.

**Comment by christianlundkvist** [comment]
@pelle I disagree, it’s not the subject of the `verReq` JWT, it’s the subject in a future different JWT. Hence I think rsub is the right terminology

**Comment by christianlundkvist** [comment]
Hmm maybe you can make the case that the request pertains to that subject in a transitive manner. I’m ok with either, but the description needs updating still.

**Comment by pelle** [comment]
@christianlundkvist I thought the description was pretty clear, but I guess not, so I've made it even clearer

**Pull request merged**